### PR TITLE
Deploy Office App through automation

### PR DESCRIPTION
**Pull request by AISDUG**
Changes to deploy office app through automation

### CODE REVIEW

**Comment by WilsonReddy** [comment]
@AISDUG Ensure source URL, Appdomain links updating correctly in office xml file prior to publishing it.

File: tree/master/cloud/src/Helper Utilities/Microsoft.Legal.MatterCenter.HelperUtilities/Microsoft.Legal.MatterCenter.UpdateAppConfig/UpdateConfig.cs [hunk 1]
```diff
@@ -332,7 +332,13 @@ public static bool UpdateXML(Dictionary<string, string> excelInput)
...
-                                                break;                                            
+                                                break;
+                                            case 3:
+                                                if (appDomain.InnerText.ToLower().Contains("sharepoint"))
+                                                {
+                                                    appDomain.InnerText = excelInput[ConstantStrings.TENANT_URL];
+                                                }
+                                                break;
                                             case 4:
                                                 appDomain.InnerText = excelInput[ConstantStrings.TENANT_URL];
```
Comment (WilsonReddy) [inline tree/master/cloud/src/Helper Utilities/Microsoft.Legal.MatterCenter.HelperUtilities/Microsoft.Legal.MatterCenter.UpdateAppConfig/UpdateConfig.cs]: Can you remove this case if it is not required?

File: tree/master/cloud/src/Helper Utilities/Microsoft.Legal.MatterCenter.HelperUtilities/Microsoft.Legal.MatterCenter.UpdateAppConfig/UpdateConfig.cs [hunk 2]
```diff
@@ -332,7 +332,13 @@ public static bool UpdateXML(Dictionary<string, string> excelInput)
                                         {
                                             case 2:
                                                 appDomain.InnerText = excelInput[ConstantStrings.AZURE_UI_SITE_URL];
-                                                break;                                            
+                                                break;
+                                            case 3:
+                                                if (appDomain.InnerText.ToLower().Contains("sharepoint"))
```
Comment (WilsonReddy) [inline tree/master/cloud/src/Helper Utilities/Microsoft.Legal.MatterCenter.HelperUtilities/Microsoft.Legal.MatterCenter.UpdateAppConfig/UpdateConfig.cs]: why you need this if condition? Remove it as it not needed.

File: tree/master/cloud/src/solution/Deployment/Scripts/Deploy-SPOFiles.ps1 [hunk 3]
```diff
@@ -11,6 +11,10 @@ Add-Type -TypeDefinition @"
 # Function to display message on console
 Function Show-Message([string] $Message, [string] $Type, [bool] $Newline = $true)
 {
+
+	# Set log file path
+$LogFile = "$ScriptDirectory\Logs\Log.txt"
```
Comment (WilsonReddy) [inline tree/master/cloud/src/solution/Deployment/Scripts/Deploy-SPOFiles.ps1]: $logfile should be set outside of this function as it user across multiple functions.  if required move it to before function declaration from bottom of the file.

File: tree/master/cloud/src/Helper Utilities/Microsoft.Legal.MatterCenter.HelperUtilities/Microsoft.Legal.MatterCenter.UpdateAppConfig/UpdateConfig.cs (hunk 2 above)
Comment (AISDUG) [inline tree/master/cloud/src/Helper Utilities/Microsoft.Legal.MatterCenter.HelperUtilities/Microsoft.Legal.MatterCenter.UpdateAppConfig/UpdateConfig.cs]: It is needed as the AppDomains are different for the Outlook app vs the Office App.

Comment (WilsonReddy) [inline tree/master/cloud/src/Helper Utilities/Microsoft.Legal.MatterCenter.HelperUtilities/Microsoft.Legal.MatterCenter.UpdateAppConfig/UpdateConfig.cs]: @AISDUG  updating site url and tenant url is same for both outlook and office.  remove 4th one and use 2nd entry for tenant url for both word and outlook app.

File: tree/master/cloud/src/solution/Deployment/Scripts/DeployOfficeApp.ps1 [hunk 4]
```diff
@@ -191,22 +192,40 @@ $ScriptDirectory = (ScriptRoot)
...
+# Set  folder paths
+$RootPath = Split-Path(Split-Path (Split-Path $PSScriptRoot -Parent) -Parent) -Parent
+$DeployPath = "$RootPath\deployments"
+$HelperPath = "$RootPath\deployments\scripts\Helper Utilities"
+$ExternalPath = "$RootPath\Helper Utilities\External"
+$DeployPath = "$RootPath\solution"
+
 #Set Excel file path, uncomment below line if you want to use this script separately
 #$ExcelFilePath = "$ParentDirectory\MCDeploymentConfig.xlsx"
+#$ExcelFilePath = "C:\Repos\mattercenter2\tree\master\cloud\src\deployments\deployments\MCDeploymentConfig.xlsx"
```
Comment (premchandpl) [inline tree/master/cloud/src/solution/Deployment/Scripts/DeployOfficeApp.ps1]: @AISDUG I think no paths should be hard coded and it should be read from config file

Comment (premchandpl) [inline tree/master/cloud/src/solution/Deployment/Scripts/DeployOfficeApp.ps1]: @AISDUG Remove all commented code if it is not necessary

**Pull request merged**